_posts/2016-07-03-handling-big-projects.markdown

```diff
@@ -0,0 +1,123 @@
+
+* Handling new issues
+* Keeping infrastructure running
+* Requests around user data
+* Keeping disparate communities together
+
+## Issues
+
+From the ground up, Danger could not end up as complex as CocoaPods, the domain they cover is different and CocoaPods sits atop of an annually moving (and _somewhat_ hostile) [platform][dev_news].
+
+However, get enough people using a product and you end up with three types of issues: Bug Reports, How Do I X? and Feature Requests.
+
+I wanted to keep bug-reports down, as much as possible, and so I built a system wherein the default error reporting system would also search GitHub issues [for similar problems][gh_inspector]. Knowing this was a generic problem, I built it with running in [other][fastlane_gh] [large][cocoapods_gh] ruby projects in mind too.
+
+`How do I X?` are issues that haven't appeared much on Danger. For CocoaPods we request people use the CocoaPods tag on StackOverflow. That saves us from 5-6 issues a day, and provides others a great place to get internet points by responding instead.
+
+Feature Requests issues are always fascinating, it gives you a chance to really see the difference between what you imagined a project's scope is, and how others perceive it. One thing that helps here, is that Danger has a [VISION.md][vision] file. This helped vocalise a lot of internal discussion, and let contributors understand the roadmap:
+
+> The core concept is that the Danger project itself creates a system that is extremely easy to build upon. The codebase for Danger should resolve specifically around systems for CI, communication with Peer Review tools and providing APIs to Source Control changes. For example: Travis CI - GitHub - git.
+
+As well as providing a heuristic for determining whether something should be added to Danger:
+
+> This means that decisions on new code integrated into Danger should ask "is this valid for every CI provider, every review system and source control type?" by making this domain so big, we can keep the core of Danger small.
+
+### Infrastructure
+
+CocoaPods has about 6 web properties, 3 of which are critical. The others can go down, or be behind the Xcode update schedules and people's projects will work fine. The 3 the critical projects are all very simple, very focused projects that only do one thing: [trunk][trunk] (provide auth, and submitting new libraries) [cocoapods.org][cocoapods_org], and [search][search]. We control everything there.
+
+Meanwhile the less critical ones like [cocoadocs.org][cocoadocs_org] have dependencies all over the show: AppleDoc, CLOC, Xcode, Cathage, Jazzy - every one of these can, and has, been a source of unreliability for infrastructure that I maintain.
+
+With Danger, I wanted to avoid building any infrastructure that does not sit on top of solid, mature projects. The website is statically created in [Middleman][middleman], using [boring][slim] [old][sass] technology, this means no server to host.
```

Comment:
I've been thinking about "boring" as a synonym for "stable", might be worth expanding on in a future post.